amico/synthesis.py

```diff
@@ -0,0 +1,638 @@
+import numpy as np
+import scipy
+from amico.util import ERROR
+
+# Limits the required precision in gpd sum
+REQUIRED_PRECISION = 1e-7
+
+# Gyromagnetic ratio
+# GAMMA = 2.675153151e8 # https://physics.nist.gov/cgi-bin/cuu/Results?search_for=proton+gyromagnetic+ratio
+# GAMMA = 2.6751525e8 # this is used in camino
+GAMMA = 2.675987e8 # this is used in NODDI
+
+def _gpd_sum(am, big_delta, small_delta, diff, radius, n):
+    sum = 0.0
+    for _am in am:
+        dam = diff * _am * _am
+        e11 = -dam * small_delta
+        e2 = -dam * big_delta
+        dif = big_delta - small_delta
+        e3 = -dam * dif
+        plus = big_delta + small_delta
+        e4 = -dam * plus
+        nom = 2 * dam * small_delta - 2 + (2 * np.exp(e11)) + (2 * np.exp(e2)) - np.exp(e3) - np.exp(e4)
+        denom = dam * dam * _am * _am * (radius * radius * _am * _am - n)
+        term = nom / denom
+        sum += term
+        if term < REQUIRED_PRECISION * sum:
+            break
+    return big_delta, small_delta, diff, radius, sum
+
+# SPHERE
+class SphereGPD():
+    AM_SPHERE = np.array([
+    2.081575978, 5.940369990, 9.205840145,
+    12.40444502, 15.57923641, 18.74264558, 21.89969648,
+    25.05282528, 28.20336100, 31.35209173, 34.49951492,
+    37.64596032, 40.79165523, 43.93676147, 47.08139741,
+    50.22565165, 53.36959180, 56.51327045, 59.65672900,
+    62.80000055, 65.94311190, 69.08608495, 72.22893775,
+    75.37168540, 78.51434055, 81.65691380, 84.79941440,
+    87.94185005, 91.08422750, 94.22655255, 97.36883035,
+    100.5110653, 103.6532613, 106.7954217, 109.9375497,
+    113.0796480, 116.2217188, 119.3637645, 122.5057870,
+    125.6477880, 128.7897690, 131.9317315, 135.0736768,
+    138.2156061, 141.3575204, 144.4994207, 147.6413080,
+    150.7831829, 153.9250463, 157.0668989, 160.2087413,
+    163.3505741, 166.4923978, 169.6342129, 172.7760200,
+    175.9178194, 179.0596116, 182.2013968, 185.3431756,
+    188.4849481, 191.6267147, 194.7684757, 197.9102314,
+    201.0519820, 204.1937277, 207.3354688, 210.4772054,
+    213.6189378, 216.7606662, 219.9023907, 223.0441114,
+    226.1858287, 229.3275425, 232.4692530, 235.6109603,
+    238.7526647, 241.8943662, 245.0360648, 248.1777608,
+    251.3194542, 254.4611451, 257.6028336, 260.7445198,
+    263.8862038, 267.0278856, 270.1695654, 273.3112431,
+    276.4529189, 279.5945929, 282.7362650, 285.8779354,
+    289.0196041, 292.1612712, 295.3029367, 298.4446006,
+    301.5862631, 304.7279241, 307.8695837, 311.0112420,
+    314.1528990
+    ])
+
+    last_big_delta = 0.0
+    last_small_delta = 0.0
+    last_diff = 0.0
+    last_radius = 0.0
+    last_sum = 0.0
+
+    def __init__(self, scheme):
+        self.scheme = scheme
+
+    def get_signal(self, diff, radius):
+        diff *= 1e-6
```

Comment:
Is this necessary or we can assume always a given units? Before there was the problem with datasynth, but now all models should use the very same convention, no?